src/components/navbar/Navbar.tsx

```diff
@@ -1,149 +1,52 @@
-            <Tooltip title="Open File">
-              <IconButton color="primary" component="span">
-                <UploadIcon />
-              </IconButton>
-            </Tooltip>
-          </label>
-
-          <Tooltip title="Save File">
-            <IconButton
-              color="primary"
-              onClick={() => {
-                this.context.saveFile();
-              }}
-            >
-              <SaveIcon />
-            </IconButton>
-          </Tooltip>
-          <Tooltip title="Export Trajectory">
-            <IconButton
-              color="primary"
-              onClick={() => {
-                this.context.exportActiveTrajectory();
-              }}
-            >
-              <FileDownload />
-            </IconButton>
-          </Tooltip>
-        </span>
-        <span
-          style={{
-            display: "flex",
-            justifyContent: "space-between",
-            paddingInline: "16px",
-          }}
-        >
-          <Divider orientation="vertical" flexItem />
-          <span
-            style={{
-              display: "flex",
-              justifyContent: "space-between",
-              paddingLeft: "16px",
-            }}
-            onClick={() => this.context.uiState.setPageNumber(0)}
-          >
-            <span
-              style={{
-                minWidth: "20rem",
-                display: "inline-block",
-                overflow: "hidden",
-                textOverflow: "ellipsis",
-                whiteSpace: "nowrap",
-                flexGrow: 1,
-                margin: "auto",
-                flexBasis: 0,
-              }}
-            >
-              {this.context.model.pathlist.activePath.name}
-            </span>
-            <Tooltip title="Open Path Dialog">
-              <IconButton color="default" className={styles.generate}>
-                <ArrowDownIcon />
-              </IconButton>
-            </Tooltip>
-          </span>
-          <Divider orientation="vertical" flexItem />
-        </span>
-        <span style={{ textAlign: "right", paddingRight: "16px" }}>
-          <Tooltip title="Field Grid">
-            <span>
-              <IconButton
-                color="primary"
-                className={styles.generate}
-                onClick={() =>
-                  this.context.uiState.setFieldGridView(
-                    !this.context.uiState.fieldGridView
-                  )
-                }
-              >
-                {this.context.uiState.fieldGridView ? (
-                  <GridIcon />
-                ) : (
-                  <GridOffIcon />
-                )}
-              </IconButton>
-            </span>
-          </Tooltip>
-          <Tooltip title="Settings">
-            <span>
-              <IconButton
-                color="primary"
-                onClick={() => this.context.uiState.setPageNumber(2)}
+          {NavbarItemData.map((item, index) => (
+            <Tooltip title={item.name}>
```

Comment:
The Tooltip component needs to have value={`${index}`}. TS will complain, but this works around a MUI bug where ToggleButtonGroup needs value props on its direct children to do highlighting of the selected option correctly.